File: lua/hamming/hamming.lua
--
-- Solution for "hamming" in Lua
-- Written by Sofia Nieves
-- Sat 28 May 06:38:26 UTC 2016
--

local hamming = {}

function hamming.compute(strand1, strand2)
  local distance = 0

  for i = 1, string.len(strand1) do
    if string.sub(strand1, i, i) ~= string.sub(strand2, i, i) then
      distance = distance + 1
    end
  end

  return distance 
end

return hamming

File: ruby/hamming/hamming.rb
#!/usr/bin/env ruby
# Solution for Hamming
# Written by Sofia Nieves
# Mon 20 Jun 05:20:42 UTC 2016
#

class Hamming
  def self.compute(strand1, strand2)
    distance = 0

    if strand1.length != strand2.length
      raise ArgumentError
    end

    for i in 0..strand1.length
      if strand1[i] != strand2[i]
        distance += 1
      end
    end

    return distance
  end
end
